codec/json/codec_test.go

```diff
@@ -0,0 +1,51 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package json
+
+import (
+	"testing"
+
+	"github.com/elastic/apm-data/model/modelpb"
+	"github.com/elastic/apm-data/model/modelpbtest"
+)
+
+func BenchmarkCodec(b *testing.B) {
+	codec := Codec{}
+	b.Run("Encode", func(b *testing.B) {
+		var output int64
+		b.RunParallel(func(p *testing.PB) {
+			for p.Next() {
+				by, _ := codec.Encode(*modelpbtest.FullEvent(b))
+				output += int64(len(by))
+			}
+		})
+
+		bytePerOp := float64(output) / float64(b.N)
+		b.ReportMetric(bytePerOp, "bytes/op")
```

Comment:
We seem to be using the same data for encoding, so it'll always be the same size per op. One thing we could do is measure per second, or report the byte per op at the top before the benchmark starts by taking `len(FullEvent)`